### ISSUE TITLE
travis: high percentage of test failures

### ISSUE THREAD

**Issue by tpepper** [01org/ciao#434]
We're having a high percentage of test failure in travis.  Maybe as high as one in four in my experience.  There is no rhyme or reason to the individual unit test failure and the test has a good chance of passing if the job is re-run.

I suspect we have a race somewhere...

**Comment by tpepper** [comment]
Could relate to https://github.com/01org/ciao/issues/83 (races) or an implicit change in the travis side that pulls in something broken (eg: https://github.com/01org/ciao/issues/420).  We need to work to isolate the issue.

**Comment by tpepper** [comment]
https://travis-ci.org/tpepper/ciao/jobs/149830418 looks like a deadlock

**Comment by tpepper** [comment]
https://travis-ci.org/01org/ciao/jobs/149835683 looks like a deadlock, and is in:

ciao-scheduler                     TestReconnects

**Comment by tpepper** [comment]
https://travis-ci.org/01org/ciao/jobs/149835681 is a straight failure, and is in:

ciao-controller                    TestTenantOutOfBounds

and outputs:

	controller_test.go:220: Not tracking limits correctly

**Comment by tpepper** [comment]
https://github.com/01org/ciao/pull/436 should address the deadlocks on the testutil channels

**Comment by tpepper** [comment]
PR https://github.com/01org/ciao/pull/437 should address the deadlocks on the testutil channels.  And it enables -race in non-controller tests to try to capture more detail information on what's breaking where.

**Comment by tpepper** [comment]
I'm closing this ticket as we've found a number of small bugs and have also chosen to disable the --race detector in our travis runs.  That should put us on a more stable footing as far as travis is concerned, but it does leave genuine issues un-resolved.

Currently there is at least one real race which @markdryan sees in the launcher output from the travis failurs.

We've got a separate ticket to enable go --race detector, which I'll leave open so we revisit this again and try to get there eventually.
